### PR TITLE
More customizable postgres container

### PR DESCRIPTION
**Pull request by Barlog-M**
Postgres docker container can be with custom database name, user name and password. Sometimes it is necessary.

### CODE REVIEW

File: modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java [hunk 1]
```diff
@@ -4,31 +4,47 @@
...
+    final String username;
+    final String password;
 
     public PostgreSQLContainer() {
-        super(IMAGE + ":latest");
+        this(IMAGE + ":latest");
     }
 
-    public PostgreSQLContainer(String dockerImageName) {
+    public PostgreSQLContainer(final String dockerImageName) {
```
Comment (bsideup) [inline modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java]: use `this(dockerImageName, "test", "test", "test")`

File: modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java [hunk 2]
```diff
@@ -4,31 +4,47 @@
  * @author richardnorth
  */
 public class PostgreSQLContainer<SELF extends PostgreSQLContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
-
     public static final String NAME = "postgresql";
     public static final String IMAGE = "postgres";
-    public static final Integer POSTGRESQL_PORT = 5432;
+    public static final Integer DEFAULT_PORT = 5432;
+    final String databaseName;
```
Comment (bsideup) [inline modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java]: make it private

File: modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java [hunk 3]
```diff
@@ -4,31 +4,47 @@
...
     @Override
     protected Integer getLivenessCheckPort() {
-        return getMappedPort(POSTGRESQL_PORT);
+        return getMappedPort(DEFAULT_PORT);
     }
 
     @Override
     protected void configure() {
-
-        addExposedPort(POSTGRESQL_PORT);
```
Comment (bsideup) [inline modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java]: please change back the order to reduce amount of changes

File: modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java [hunk 4]
```diff
@@ -39,17 +55,17 @@ public String getDriverClassName() {
 
     @Override
     public String getJdbcUrl() {
-        return "jdbc:postgresql://" + getContainerIpAddress() + ":" + getMappedPort(POSTGRESQL_PORT) + "/test";
+        return "jdbc:postgresql://" + this.getContainerIpAddress() + ":" + this.getMappedPort(DEFAULT_PORT) + "/" + databaseName;
     }
 
     @Override
     public String getUsername() {
```
Comment (bsideup) [inline modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java]: use Lombok's `@Getter`

File: modules/postgresql/src/test/java/org/testcontainers/junit/CustomizablePostgreSQLTest.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Rule
+    public PostgreSQLContainer postgres = new PostgreSQLContainer(
+        "postgres:latest", DB_NAME, USER, PWD);
+
+    @Test
+    public void testSimple() throws SQLException {
+        HikariConfig hikariConfig = new HikariConfig();
+        hikariConfig.setJdbcUrl(postgres.getJdbcUrl());
+        hikariConfig.setUsername(USER);
```
Comment (bsideup) [inline modules/postgresql/src/test/java/org/testcontainers/junit/CustomizablePostgreSQLTest.java]: `postgres.getUsername()` ?

File: modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java [hunk 6]
```diff
@@ -4,31 +4,47 @@
  * @author richardnorth
  */
 public class PostgreSQLContainer<SELF extends PostgreSQLContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
-
     public static final String NAME = "postgresql";
     public static final String IMAGE = "postgres";
-    public static final Integer POSTGRESQL_PORT = 5432;
```
Comment (bsideup) [inline modules/postgresql/src/main/java/org/testcontainers/containers/PostgreSQLContainer.java]: please avoid nun-functional changes

**Comment by rnorth** [comment]
Looks good to me - just one thing. I've generally erred away from constructor args for containers (apart from specifying container image name), and gone for fluent setters instead.

Is there a reason why the fluent setter pattern can't be used here? I'd prefer to be as consistent as possible...

**Pull request merged**